### PR TITLE
fix: add mfaSettings in initOptions

### PR DESCRIPTION
**Pull request by grvgoel81**
fix: add mfaSettings in initOptions

### CODE REVIEW

File: core/src/main/java/com/web3auth/core/Web3Auth.kt [hunk 1]
```diff
@@ -44,11 +44,14 @@ class Web3Auth(web3AuthOptions: Web3AuthOptions) {
             gson.toJson(web3AuthOption.loginConfig)
         if (web3AuthOption.buildEnv != null) initOptions["buildEnv"] =
             web3AuthOption.buildEnv.toString().lowercase()
+        if (web3AuthOption.mfaSettings != null) initOptions["mfaSettings"] =
```
Comment (chaitanyapotti) [inline core/src/main/java/com/web3auth/core/Web3Auth.kt:47]: pls also add sessionTime to initOptions

File: core/src/main/java/com/web3auth/core/Web3Auth.kt [hunk 2]
```diff
@@ -44,11 +44,14 @@ class Web3Auth(web3AuthOptions: Web3AuthOptions) {
...
+            gson.toJson(web3AuthOption.mfaSettings)
 
         val initParams = mutableMapOf(
             "loginProvider" to params?.loginProvider,
             "extraLoginOptions" to params?.extraLoginOptions,
-            "redirectUrl" to if (params?.redirectUrl != null) params.redirectUrl.toString() else initOptions["redirectUrl"].toString()
+            "redirectUrl" to if (params?.redirectUrl != null) params.redirectUrl.toString() else initOptions["redirectUrl"].toString(),
+            "mfaLevel" to params?.mfaLevel
```
Comment (chaitanyapotti) [inline core/src/main/java/com/web3auth/core/Web3Auth.kt:54]: Also, pass curve, dappShare here

File: core/src/main/java/com/web3auth/core/types/LoginConfigItem.kt [hunk 3]
```diff
@@ -5,7 +5,7 @@ data class LoginConfigItem(
     private var typeOfLogin: TypeOfLogin,
     private var name: String? = null,
     private var description: String? = null,
-    private var clientId: String,
+    private var clientId: String? = null,
```
Comment (chaitanyapotti) [inline core/src/main/java/com/web3auth/core/types/LoginConfigItem.kt:8]: this is mandatory now

Comment (grvgoel81) [reply core/src/main/java/com/web3auth/core/types/LoginConfigItem.kt:8]: its mandatory in Web3AuthOptions.

Comment (chaitanyapotti) [reply core/src/main/java/com/web3auth/core/types/LoginConfigItem.kt:8]: this clientId is the OAuth client id.. like for auth0/firebase client id.
web3auth options client id is from our dashboard

**Comment by chaitanyapotti** [comment]
sessionTime must be in initOptions, not initParams

File: core/src/main/java/com/web3auth/core/types/LoginConfigItem.kt (hunk 3 above)
Comment (grvgoel81) [reply core/src/main/java/com/web3auth/core/types/LoginConfigItem.kt:8]: ok

File: core/src/main/java/com/web3auth/core/Web3Auth.kt [hunk 4]
```diff
@@ -44,11 +44,18 @@ class Web3Auth(web3AuthOptions: Web3AuthOptions) {
...
+            params.sessionTime.toString()
 
         val initParams = mutableMapOf(
             "loginProvider" to params?.loginProvider,
             "extraLoginOptions" to params?.extraLoginOptions,
-            "redirectUrl" to if (params?.redirectUrl != null) params.redirectUrl.toString() else initOptions["redirectUrl"].toString()
+            "redirectUrl" to if (params?.redirectUrl != null) params.redirectUrl.toString() else initOptions["redirectUrl"].toString(),
+            "mfaLevel" to params?.mfaLevel,
+            "curve" to params?.curve,
+            "dappShare" to params?.dappShare
```
Comment (chaitanyapotti) [inline core/src/main/java/com/web3auth/core/Web3Auth.kt:58]: read the dapp share from storage..

Comment (grvgoel81) [reply core/src/main/java/com/web3auth/core/Web3Auth.kt:58]: before calling request() function in login(), we are reading dapp share from storage like below:

if (web3AuthOption.loginConfig != null) {
            val loginConfigItem: LoginConfigItem? = web3AuthOption.loginConfig?.values?.first()
            val share: String? =
                KeyStoreManagerUtils.decryptData(loginConfigItem?.verifier.toString())
            if (share?.isNotEmpty() == true) {
                loginParams.dappShare = share
            }
        }

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Pull request merged**